sklearn/decomposition/pca.py

```diff
@@ -251,50 +303,68 @@ def fit_transform(self, X, y=None):
         return U
 
     def _fit(self, X):
-        """Fit the model on X
+        """Dispatch the actual fitting to _fit_full and _fit_truncated, after
+        handling svd_solver='auto' policy.
+        """
+        X = check_array(X, dtype=[np.float32, np.float64], ensure_2d=True,
+                        copy=self.copy)
 
-        Parameters
-        ----------
-        X: array-like, shape (n_samples, n_features)
-            Training vector, where n_samples in the number of samples and
-            n_features is the number of features.
+        # Handle n_components==None
+        if self.n_components is None:
+            n_components = X.shape[1]
+        else:
+            n_components = self.n_components
 
-        Returns
-        -------
-        U, s, V : ndarrays
-            The SVD of the input data, copied and centered when
-            requested.
+        # Handle svd_solver
+        svd_solver = self.svd_solver
+        if svd_solver == 'auto':
+            if n_components < .8 * min(X.shape):
```

Comment:
It should be:

```python
if n_components >=1 and n_components < .8 * min(X.shape):
```

instead. `n_components in [0, 1]` means using the explained variance ratio to select the actual number of components once a full SVD has been performed.